### PR TITLE
Migrate e2e tests from CoreOS Container Linux to Fedora CoreOS

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

* Migrate e2e tests from CoreOS Container Linux to Fedora CoreOS;
* Upgrade the version of kubernetes because the old version we were using is not compatible anymore with the version of systemd shipped with Fedora CoreOS (The kubelet failed to control cgroup);
* Upgrade the version of Argo because the old version we were using was using beta versions of kubernetes APIs which are not available anymore;
* Update the workflow to make it compatible with the latest version of Argo and leverage `WorkflowTemplate`s to split it in several files.

### Motivation

CoreOS Container Linux is EOL and the AWS AMIs have been removed so that the e2e tests cannot be launched anymore.

### Describe your test plan

Trigger the e2e tests job in the CI pipeline.

### CODE REVIEW

File: test/e2e/argo-workflows/agent/templates/fake-datadog.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                maxSurge: 1
+                maxUnavailable: 0
+            template:
+              metadata:
+                labels:
+                  app: fake-datadog
+              spec:
+                containers:
+                - name: api
+                  image: l3n41c/fake-datadog:latest
```
Comment (L3n41c) [inline test/e2e/argo-workflows/agent/templates/fake-datadog.yaml:62]: Remember to publish a new `datadog/fake-datadog` image and update here.

**Comment by L3n41c** [comment]
Here is a successful `e2e` job from GitLab: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/42705455

Is is a graphical view of the Argo workflow:
image

File: test/e2e/containers/fake_datadog/app/api.py [hunk 2]
```diff
@@ -69,14 +69,31 @@ def record_and_loads(filename: str, content_type: str, content_encoding: str, co
...
+    elif isinstance(data, list):
+        return [patch_data(i, patch_key, patch_leaf) for i in data]
+    else:
+        return patch_leaf(data)
+
+
+def fix_data(data):
+    return patch_data(
+        data,
+        patch_key=lambda x: x.translate(str.maketrans('.$', '\uff0e\uff04')),
```
Comment (vboulineau) [inline test/e2e/containers/fake_datadog/app/api.py:84]: That's a bit sneaky :D, why not replacing with more common chars like `_`

File: test/e2e/scripts/run-instance/23-argo-get.sh [hunk 3]
```diff
@@ -1,42 +1,54 @@
...
+done
+
+# Make the Argo UI available from the user
+/opt/bin/kubectl patch svc -n argo argo-server --type json --patch $'[{"op": "replace", "path": "/spec/type", "value": "NodePort"}]'
+
+until [[ -n ${KUBE_SVC:+x} ]]; do
+    sleep 1
+    KUBE_SVC="$(sudo iptables -w -t nat -L KUBE-NODEPORTS -n -v | awk '/argo\/argo-server:web/ && $3 ~ /^KUBE-SVC-/ {print $3}')"
 done
+sudo iptables -w -t nat -N HACK
```
Comment (vboulineau) [inline test/e2e/scripts/run-instance/23-argo-get.sh:43]: This block requires documentation

File: test/e2e/containers/fake_datadog/app/api.py (hunk 2 above)
Comment (L3n41c) [reply test/e2e/containers/fake_datadog/app/api.py:84]: Well, I just found many StackOverflow or blog posts pointing to the [official MongoDB documentation](https://docs.mongodb.com/v2.6/faq/developers/#dollar-sign-operator-escaping) which says:
> In some cases, you may wish to build a BSON object with a user-provided key. In these situations, keys will need to substitute the reserved $ and . characters. Any character is sufficient, but consider using the Unicode full width equivalents: U+FF04 (i.e. “＄”) and U+FF0E (i.e. “．”).

So, doing this seemed to be the MongoDB idiomatic way of dealing with this issue.
But `_` would have definitively do the job as well.

Comment (vboulineau) [reply test/e2e/containers/fake_datadog/app/api.py:84]: Fair enough, just put that in comment then

**Review by vboulineau — APPROVED** [review]

File: test/e2e/argo-workflows/agent/templates/fake-datadog.yaml (hunk 1 above)
Comment (L3n41c) [reply test/e2e/argo-workflows/agent/templates/fake-datadog.yaml]: Done !

**Review by KSerrania — APPROVED** [review]
The CI / invoke changes look good to me.

**Pull request merged**